### PR TITLE
added callback function in uzfs_get_txg_diff_tree

### PR DESCRIPTION
**Pull request by mynktl**
Changes in uzfs_get_txg_diff_tree API
This API will traverse zvol and execute callback function provided by caller with offset/len/blockid.

### CODE REVIEW

File: lib/libzpool/uzfs_mtree.c [hunk 1]
```diff
@@ -211,10 +211,9 @@ uzfs_txg_diff_tree_compare(const void *arg1, const void *arg2)
 	return (AVL_CMP(node1->offset, node2->offset));
 }
 
-
 int
 uzfs_get_txg_diff_tree(zvol_state_t *zv, uint64_t start_txg, uint64_t end_txg,
-    avl_tree_t **tree)
+    uzfs_zvol_traverse_t *func, void *arg)
```
Comment (vishnuitta) [inline lib/libzpool/uzfs_mtree.c]: uzfs_txg_diff_traverse_cb_t

File: include/uzfs_mtree.h [hunk 2]
```diff
@@ -22,10 +22,30 @@
 #ifndef	_UZFS_MTREE_H
 #define	_UZFS_MTREE_H
 
+/*
+ * API to get modified blocks between start_txg and end_txg
+ * Note: API will alocate condensed tree and populate it witch
```
Comment (vishnuitta) [inline include/uzfs_mtree.h]: update the comments

File: lib/libzpool/uzfs_mtree.c (hunk 1 above)
Comment (jkryl) [inline lib/libzpool/uzfs_mtree.c]: I think we should remove "tree" from function name for better clarity.

**Review by jkryl — APPROVED** [review]

File: lib/libzpool/uzfs_mtree.c (hunk 1 above)
Comment (mynktl) [reply lib/libzpool/uzfs_mtree.c]: ok

File: include/uzfs_mtree.h (hunk 2 above)
Comment (mynktl) [reply include/uzfs_mtree.h]: ok

File: lib/libzpool/uzfs_mtree.c (hunk 1 above)
Comment (mynktl) [reply lib/libzpool/uzfs_mtree.c]: ok

**Pull request merged**